### PR TITLE
Pr 113 metadata err msg

### PR DESCRIPTION
**Pull request by jhiemstrawisc**
This one fell through the cracks for me, but I'm taking over PR #113 to handle failed CI test and linter issues.

### CODE REVIEW

**Comment by djw8605** [comment]
Could we do something as simple as just assigning the err_msg to a shared/unique ptr?

File: src/scitokens_internal.cpp [hunk 1]
```diff
@@ -84,6 +88,20 @@ SimpleCurlGet::GetStatus SimpleCurlGet::perform_start(const std::string &url) {
     return perform_continue();
 }
 
+std::string SimpleCurlGet::get_url() const {
+    if (!m_curl) {
+        return "";
+    }
+
+    char *url = nullptr;
+    auto rv = curl_easy_getinfo(m_curl.get(), CURLINFO_EFFECTIVE_URL, &url);
```
Comment (djw8605) [inline src/scitokens_internal.cpp:97]: Do we need to free the url memory?

**Review by djw8605 — COMMENTED** [review]
A single comment.

File: src/scitokens_internal.cpp (hunk 1 above)
Comment (jhiemstrawisc) [reply src/scitokens_internal.cpp:97]: It appears no. Check the last sentence of `Description` from https://linux.die.net/man/3/curl_easy_getinfo

**Pull request merged**